### ISSUE TITLE
 `useNamingConvention`: support uppercase function names for HTTP verbs

### ISSUE THREAD

**Issue by kevinwolfcr** [biomejs/biome#2314]
### Environment information

```bash
CLI:
  Version:                      1.6.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.0.35"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Linter:
  Recommended:                  false
  All:                          true
  Rules:                        nursery/useSortedClasses = {"level":"error","options":{"attributes":["class","className","className","classList"],"functions":["clsx","cva","tw","cn"]}}

Workspace:
  Open Documents:               0
```

### Rule name

useNamingConvention

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABHAEUAVAAoACkAIAB7AAoAIAAgAC8ALwAgAAoAfQA%3D

### Expected result

Some frameworks like Next.js [allows creating API endpoints](https://nextjs.org/docs/app/building-your-application/routing/route-handlers). For that you should export functions using any HTTP verb, in uppercase mode. I think they could be allowed on this rule.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
I am not sure that this should be allowed by efault because it is a specific usage.

This could be addressed via configuration?
We have an open issue #1900 for adding some options to `useNamingConvention`.

**Comment by ematipico** [comment]
> I am not sure that this should be allowed by default because it is a specific usage.
> 

Unfortunately, no. Other frameworks are using this pattern: SvelteKit and Astro

**Comment by Conaclos** [comment]
NextJS [supports](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#supported-http-methods) `GET`, `POST`, `PUT`, `PATCH`, `DELETE`, `HEAD`, and `OPTIONS`.

Astro seems to [support](https://docs.astro.build/en/guides/endpoints/#http-methods) any [HTTP methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods).

I need to think about it. I am open to any suggestion to support this pattern.

**Comment by kevinwolfcr** [comment]
Maybe allow only the HTTP verbs?
